libraries/arrow-convert/src/from_impls.rs

```diff
@@ -163,6 +167,237 @@ impl<'a> TryFrom<&'a ArrowData> for Vec<u8> {
+    type Error = eyre::Report;
+    fn try_from(value: &'a ArrowData) -> Result<Self, Self::Error> {
+        let array: &PrimitiveArray<arrow::datatypes::Int32Type> = value
+            .as_primitive_opt()
+            .wrap_err("not a primitive Int32Type array")?;
+        if array.null_count() != 0 {
+            eyre::bail!("array has nulls");
+        }
+        Ok(array.values())
+    }
+}
+
+impl<'a> TryFrom<&'a ArrowData> for Vec<i32> {
+    type Error = eyre::Report;
+    fn try_from(value: &'a ArrowData) -> Result<Self, Self::Error> {
+        value.try_into().map(|slice: &'a [i32]| slice.to_vec())
+    }
+}
+
+impl<'a> TryFrom<&'a ArrowData> for &'a [i64] {
+    type Error = eyre::Report;
+    fn try_from(value: &'a ArrowData) -> Result<Self, Self::Error> {
+        let array: &PrimitiveArray<arrow::datatypes::Int64Type> = value
+            .as_primitive_opt()
+            .wrap_err("not a primitive Int64Type array")?;
+        if array.null_count() != 0 {
+            eyre::bail!("array has nulls");
+        }
+        Ok(array.values())
+    }
+}
+
+impl<'a> TryFrom<&'a ArrowData> for Vec<i64> {
+    type Error = eyre::Report;
+    fn try_from(value: &'a ArrowData) -> Result<Self, Self::Error> {
+        value.try_into().map(|slice: &'a [i64]| slice.to_vec())
+    }
+}
+
+impl<'a> TryFrom<&'a ArrowData> for &'a [f32] {
+    type Error = eyre::Report;
+    fn try_from(value: &'a ArrowData) -> Result<Self, Self::Error> {
+        let array: &PrimitiveArray<arrow::datatypes::Float32Type> = value
+            .as_primitive_opt()
+            .wrap_err("not a primitive Float32Type array")?;
+        if array.null_count() != 0 {
+            eyre::bail!("array has nulls");
+        }
+        Ok(array.values())
+    }
+}
+
+impl<'a> TryFrom<&'a ArrowData> for Vec<f32> {
+    type Error = eyre::Report;
+    fn try_from(value: &'a ArrowData) -> Result<Self, Self::Error> {
+        value.try_into().map(|slice: &'a [f32]| slice.to_vec())
+    }
+}
+
+impl<'a> TryFrom<&'a ArrowData> for &'a [f64] {
+    type Error = eyre::Report;
+    fn try_from(value: &'a ArrowData) -> Result<Self, Self::Error> {
+        let array: &PrimitiveArray<arrow::datatypes::Float64Type> = value
+            .as_primitive_opt()
+            .wrap_err("not a primitive Float64Type array")?;
+        if array.null_count() != 0 {
+            eyre::bail!("array has nulls");
+        }
+        Ok(array.values())
+    }
+}
+
+impl<'a> TryFrom<&'a ArrowData> for Vec<f64> {
+    type Error = eyre::Report;
+    fn try_from(value: &'a ArrowData) -> Result<Self, Self::Error> {
+        value.try_into().map(|slice: &'a [f64]| slice.to_vec())
+    }
+}
+
+impl<'a> TryFrom<&'a ArrowData> for String {
+    type Error = eyre::Report;
+    fn try_from(value: &'a ArrowData) -> Result<Self, Self::Error> {
+        let string_array: Vec<String> =
+            <Arc<dyn arrow::array::Array> as Clone>::clone(&value).try_into_collection()?;
+        return Ok(string_array[0].clone());
```

Comment:
Can we make a check to not panic if the length of vec is actually 0 for whatever reason?